### ISSUE TITLE
MueLu: unit test for indices larger than 2.1b is failing

### ISSUE THREAD

**Issue by jhux2** [trilinos/Trilinos#2536]
The test is MueLu_TwoBillionTest.exe.  Failure can be found [here](https://testing.sandia.gov/cdash/buildSummary.php?buildid=3493527).

@trilinos/muelu @trilinos/zoltan2

**Comment by jhux2** [comment]
<details>
<summary>Test Output</summary>

```
TwoBillion: Running Test
pid 0 : 1st GID = 3000000000
pid 2 : 1st GID = 3000002000
pid 1 : 1st GID = 3000001000
pid 3 : 1st GID = 3000003000
Clearing old data (if any)
Level 0
 Setup Smoother (MueLu::Ifpack2Smoother{type = RELAXATION})
  "Ifpack2::Relaxation": {Initialized: true, Computed: true, Type: Symmetric Gauss-Seidel, sweeps: 2, damping factor: 1, Global matrix dimensions: [4000, 4000], Global nnz: 11998}
Level 1
 Build (MueLu::RebalanceTransferFactory)
  Build (MueLu::RepartitionFactory)
   Computing Ac (MueLu::RAPFactory)
    Prolongator smoothing (MueLu::SaPFactory)
     Matrix filtering (MueLu::FilteredAFactory)
      Build (MueLu::CoalesceDropFactory)
       Build (MueLu::AmalgamationFactory)
       lightweight wrap = 1
       algorithm = "classical": threshold = 0, blocksize = 1
       Detected 0 Dirichlet nodes
       Number of dropped entries in unamalgamated matrix graph: 0/11998 (0%)
      Filtered matrix is not being constructed as no filtering is being done
     Build (MueLu::TentativePFactory)
      Build (MueLu::UncoupledAggregationFactory)
       BuildAggregates (Phase - (Dirichlet))

p=3: *** Caught standard std::exception of type 'std::logic_error' :

 /space/nightlyTesting/OPENMPI-1.10.1_DEBUG_LONG-LONG/Trilinos/packages/teuchos/core/src/Teuchos_ArrayRCP.hpp:1155:
 
 Throw number = 24
 
 Throw test that evaluated to true: !( (lowerOffset_ <= lowerOffset_in && lowerOffset_in+size_in-1 <= upperOffset_) && size_in >= 0 )
 
 Teuchos::ArrayRCP<int>::assert_in_range: Error, [lowerOffset,lowerOffset+size-1] = [0,0] does not lie in the range [0,-1]!

p=1: *** Caught standard std::exception of type 'std::logic_error' :

 /space/nightlyTesting/OPENMPI-1.10.1_DEBUG_LONG-LONG/Trilinos/packages/teuchos/core/src/Teuchos_ArrayRCP.hpp:1155:
 
 Throw number = 24
 
 Throw test that evaluated to true: !( (lowerOffset_ <= lowerOffset_in && lowerOffset_in+size_in-1 <= upperOffset_) && size_in >= 0 )
 
 Teuchos::ArrayRCP<int>::assert_in_range: Error, [lowerOffset,lowerOffset+size-1] = [0,0] does not lie in the range [0,-1]!

p=2: *** Caught standard std::exception of type 'std::logic_error' :

         aggregated : 0 (phase), 0/4000 [0.00%] (total)
         remaining  : 4000
         aggregates : 0 (phase), 0 (total)
       BuildAggregates (Phase 1 (main))
         aggregated : 4000 (phase), 4000/4000 [100.00%] (total)
         remaining  : 0
         aggregates : 1336 (phase), 1336 (total)
       BuildAggregates (Phase 2a (secondary))
         aggregated : 0 (phase), 4000/4000 [100.00%] (total)
         remaining  : 0
         aggregates : 0 (phase), 1336 (total)
       BuildAggregates (Phase 2b (expansion))
         aggregated : 0 (phase), 4000/4000 [100.00%] (total)
         remaining  : 0
         aggregates : 0 (phase), 1336 (total)
       BuildAggregates (Phase 3 (cleanup))
         aggregated : 0 (phase), 4000/4000 [100.00%] (total)
         remaining  : 0
         aggregates : 0 (phase), 1336 (total)
       "UC": MueLu::Aggregates{nGlobalAggregates = 1336}
      Build (MueLu::CoarseMapFactory)
     Calculating max eigenvalue estimate now (max iters = 10)
     Prolongator damping factor = 0.70 (1.33 / 1.89)
 /space/nightlyTesting/OPENMPI-1.10.1_DEBUG_LONG-LONG/Trilinos/packages/teuchos/core/src/Teuchos_ArrayRCP.hpp:1155:
 
 Throw number = 24
 
 Throw test that evaluated to true: !( (lowerOffset_ <= lowerOffset_in && lowerOffset_in+size_in-1 <= upperOffset_) && size_in >= 0 )
 
 Teuchos::ArrayRCP<int>::assert_in_range: Error, [lowerOffset,lowerOffset+size-1] = [0,0] does not lie in the range [0,-1]!
    Transpose P (MueLu::TransPFactory)
   RAPFactory: call transfer factory: MueLu::CoordinatesTransferFactory
   Build (MueLu::CoordinatesTransferFactory)
    Transferring coordinates
   Build (MueLu::RepartitionHeuristicFactory)
    Repartitioning? YES:
      min # rows per proc = 334, min allowable = 400
    Number of partitions to use = 3
   Build (MueLu::Zoltan2Interface)
    Checking consistence of row and coordinates maps
    Zoltan2 parameters:
    ----------
    algorithm = multijagged   [unused]
    partitioning_approach = partition   [unused]
    mj_premigration_option = 1   [unused]
    num_global_parts = 3   [unused]
```
</details>

**Comment by mhoemmen** [comment]
Fixed in develop by PR #2537, just merged.
